### DISCUSSION TITLE
Focus navigation within Data Grids when Cells have multiple focusable elements

### DISCUSSION THREAD

**Discussion by dholmes2409** [w3c/wcag#3678] [Q&A]
I want to improve the keyboard navigation of my Data Grid and in some scenarios I would like to support having multiple focusable elements within a single cell. The examples at https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/ only include at most 1 focusable element within a cell and when navigating to that cell the focus is automatically given to that element but if I were to have multiple:

```html
<table>
  ...
  <tr>
    ...
    <td>
      <a tabindex="-1" href="#">Cash Deposit</a>
      <a tabindex="-1" href="#">Cash Withdrawal</a>
    </td>
    ...
  </tr>
<table>
```

What would people advise to use for navigation?

**Comment by JAWS-test** [comment]
https://www.w3.org/WAI/ARIA/apg/patterns/grid/#keyboardinteraction-settingfocusandnavigatinginsidecells:

After deactivating the grid navigation, navigation within the cell takes place using the usual methods (e.g. TAB from control to control , with the arrow keys within selection lists or radiobutton groups). Navigation remains limited to the current cell.

**Reply by dholmes2409** [reply]
That makes sense, thank you for the prompt reply! 
To further add to this, if I were to make the entire row clickable to invoke a behaviour (for example, opening a modal with further detail) would this require a different focus navigation behaviour? For example using Up/Down arrow keys to navigate rows, and then Left/Right to navigate cells within _currently_ focused row but moving Up/Down would then focus on the entire row Above/Below. Or would you advise against this and implement this differently?

```tsx
<table>
  ...
  <tr onClick={openModal}>
    ...
    <td>
      <a tabindex="-1" href="#">Cash Deposit</a>
      <a tabindex="-1" href="#">Cash Withdrawal</a>
    </td>
    ...
  </tr>
<table>
```

**Reply by JAWS-test** [reply]
Row-by-row navigation in the grid does not make sense for various reasons:

* it is difficult for all users to recognize that a row can be activated as a whole (usability problem)
* keyboard users find it difficult to recognize that they cannot navigate vertically and horizontally cell by cell as expected (problem with SC 2.1.1)
* Blind users cannot recognize that the entire row serves as a control element (problem with SC 4.1.2)
* the label of the row would be its entire content, which would not be a meaningful label for voice output (problem with SC 2.4.6)
* the row could not be activated with voice control either.

For the functionality, there should be a button in exactly one cell of the grid in each row with which the action can be explicitly executed.
